### ISSUE TITLE
How to pick the best result model file when I run distributed training?

### ISSUE THREAD

**Issue by swlsw** [dmlc/mxnet#1277]
We run distributed training for ImageNet(inception-bn model) using 8~16 machines.
After finishing the training, several model files are created in each machine.
Then how can I pick the best model file?
Manually with log file? or other method?

**Comment by piiswrong** [comment]
models with the same epoch number should be the same.

**Comment by swlsw** [comment]
Thanks for your response.
Then why does it have different validation accuracy at same epoch between machines?
If partitioned val data causes that result, how can I know the validation accuracy for entire val data?
I'm wondering if there is some args for using partitioned val data or entire val data.

**Comment by antinucleon** [comment]
Because in BatchNorm network, moving-average/moving-variance is not synced. Different model has different moving parameters. You are supposed to update the moving parameter at the very end of training.

**Comment by swlsw** [comment]
Thanks for your response about BatchNorm.
